pkg/controller/controller.go

```diff
@@ -94,12 +98,13 @@ type ReQueue interface {
 }
 
 // NewQueueRetryManager safely creates a new QueueRetryManager.
-func NewQueueRetryManager(queue ReQueue, keyFn kcache.KeyFunc, retryFn RetryFunc) *QueueRetryManager {
+func NewQueueRetryManager(queue ReQueue, keyFn kcache.KeyFunc, retryFn RetryFunc, rate float32, burst int) *QueueRetryManager {
 	return &QueueRetryManager{
 		queue:     queue,
 		keyFunc:   keyFn,
 		retryFunc: retryFn,
 		retries:   make(map[string]int),
+		limiter:   kutil.NewTokenBucketRateLimiter(rate, burst),
```

Comment:
An example here - if we want to add a new rate limited token behavior, we shouldn't have to take more arguments to QueueRetryManager.